### DISCUSSION TITLE
Add full resolution option to encoder

### DISCUSSION THREAD

**Discussion by zygmund2000** [PE5PVB/TEF6686_ESP32#279] [Ideas]
Hi,

At the moment encoder looks like half resolution behavior, it's one of possible solutions and would very nice to add full as option to choose in menu.

Cheers

**Comment by PE5PVB** [comment]
What do you mean?

**Comment by zygmund2000** [comment]
It's mean if rotating encoder eg. frequency changes not every step of encoder but after 2 steps, so 1 step do nothing. Old software had changes every single step.
Encoders like that can be managed by 2 methods, eg if rotating one direction gives you 2->3->1->0->2 etc and that it is full resolution, the second method involves detecting the falling edge of the signal on one of the encoder pins and then checking the status of the other pin.
So now it looks like second method.

**Comment by PE5PVB** [comment]
Software is build for the cheap dirty encoders with the known glitches. So I wait untill both pins get high before action is done. To disable this you can push the encoder while booting. This can be helpfull when using an optical encoder.

I don't think it's usefull to add more options, because normal users can get confused about this. If you build your own radio with another type of encoder, please just change the code yourself.
